Lib/defconQt/glyphView.py

```diff
@@ -217,34 +333,200 @@ def __init__(self, glyph, parent=None):
 
         self.setContextMenuPolicy(Qt.ActionsContextMenu)
         createAnchorAction = QAction("Add Anchor…", self)
-        createAnchorAction.triggered.connect(self.view.createAnchor)
+        createAnchorAction.triggered.connect(self._redirect('view', 'createAnchor'))
         self.addAction(createAnchorAction)
         createComponentAction = QAction("Add Component…", self)
-        createComponentAction.triggered.connect(self.view.createComponent)
+        createComponentAction.triggered.connect(self._redirect('view', 'createComponent'))
         self.addAction(createComponentAction)
 
-        self.setCentralWidget(self.view)
+        for layer in self._layerSet:
+            self._listenToLayer(layer)
+
+        self._changeGlyph(glyph)
+        self._updateComboBox()
+
         self.setWindowTitle(glyph.name, glyph.getParent())
         self.adjustSize()
 
+    def _changeGlyph(self, glyph):
+        oldView = self.view
+        if oldView:
+            # Preserve the selected layer (by setting the glyph from that layer)
+            # Todo: If that layer is already in the glyph, it would be a little bit
+            # harder to create it here and may be better or worse. Worse becaue
+            # we'd alter the data without being explicitly asked to do so.
+            # Ask someone who does UX.
+            # ToDo: it's a bit strange to have the view store the current layer
+            # but this is the only case where it happens to be strange.
+            if oldView.layer is not glyph.layer and glyph.name in oldView.layer:
+                glyph = oldView.layer[glyph.name]
+            oldView.hide()
+            oldView.deleteLater()
+
+        self.view = GlyphView(glyph, self._displaySettings, self)
+        # Preserve the zoom level of the oldView.
+        # FIXME: is there more state that we need to carry over to the next
+        # GlyphView? (This seems pretty complete to me…) If yes, we should
+        # make an interface for a predecessor -> successor transformation
+        if oldView:
+            self.view.setTransform(oldView.transform())
```

Comment:
I feel that e.g. what happens in this function is making things more complex and less straightforward to reason about. What is the benefit for this downside?

Why creating a new GlyphView rather than redrawing things inside of it?